http/src/config.rs

```diff
@@ -245,6 +275,7 @@ struct CompatibleConfigFile {
 #[serde(rename_all = "PascalCase")]
 struct Addresses {
     swarm: Vec<Multiaddr>,
+    api: SocketAddr,
```

Comment:
I looked over an older go-ipfs config file which only seems to support a single API listening address *in multiaddr*. If this hasn't changed I'd rename all variables touching this `Addresses.api` as singular from plurals.